### PR TITLE
lock-stall-defconfig: add basic stuff to debug locks/hangs

### PR DESCRIPTION
**Pull request by plbossart**
Settings shamelessly borrowed from

https://stackoverflow.com/questions/20892822/how-to-use-lockdep-feature-in-linux-kernel-for-deadlock-detection

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Comment by plbossart** [comment]
@bardliao @kv2019i @ranj063 can ask you for a test on your device (SoundWire/HDaudio/Volteer respectively) ?

I saw some issues on SoundWire with sysfs stuff, I am pretty sure that set of kconfig will expose a number of issues in our code...

**Comment by ranj063** [comment]
> @bardliao @kv2019i @ranj063 can ask you for a test on your device (SoundWire/HDaudio/Volteer respectively) ?
> 
> I saw some issues on SoundWire with sysfs stuff, I am pretty sure that set of kconfig will expose a number of issues in our code...

@plbossart ack. Is there a particular test you're interested in or is it just the usual run-all-tests that I should run?

**Comment by bardliao** [comment]
I am running run-all-tests test on my SoundWire device.

**Comment by bardliao** [comment]
No regression found on my CML SoundWire device

**Comment by plbossart** [comment]
@ranj063 yes standard tests. I am interested in the SoundWire bus lock, as well as all the ASoC checks that we currently don't enable.

File: kconfig-sof-default.sh [hunk 1]
```diff
@@ -27,4 +27,7 @@ $COMMAND .config \
 	 $KCONFIG_DIR/mach-driver-defconfig \
 	 $KCONFIG_DIR/hdaudio-codecs-defconfig \
 	 $KCONFIG_DIR/telemetry-debugfs-defconfig \
+	 $KCONFIG_DIR/lock-stall-defconfig \
 	 $KCONFIG_DIR/soundwire-defconfig $@
+
```
Comment (fredoh9) [inline kconfig-sof-default.sh:32]: Two empty lines.

File: lock-stall-defconfig [hunk 2]
```diff
@@ -0,0 +1,23 @@
+CONFIG_LOCKUP_DETECTOR=y
+
+CONFIG_HARDLOCKUP_DETECTOR=y
+CONFIG_HARDLOCKUP_DETECTOR_PERF=y
+
+CONFIG_SOFTLOCKUP_DETECTOR=y
+
+CONFIG_DETECT_HUNG_TASK=y
+
```
Comment (fredoh9) [inline lock-stall-defconfig:9]: Two empty lines

**Review by fredoh9 — COMMENTED** [review]
We don't need this for nocodec?

File: kconfig-sof-default.sh (hunk 1 above)
Comment (plbossart) [reply kconfig-sof-default.sh]: fixed

File: lock-stall-defconfig (hunk 2 above)
Comment (plbossart) [reply lock-stall-defconfig:9]: fixed, thanks

**Comment by plbossart** [comment]
> We don't need this for nocodec?

we might as well add it, yes. Thanks for the suggestion @fredoh9. Pushed now.

**Pull request merged**